account_financial_report/models/account_group.py

```diff
@@ -37,6 +41,44 @@ def _compute_complete_name(self):
             else:
                 group.complete_name = group.name
 
+    @api.depends("code_prefix_start", "code_prefix_end")
+    def _compute_account_ids(self):
+        """Retrieves every account from `self`.
+        In Odoo 18 the group_id on account is not stored so it raises
+        an error the one2many account_ids with inverse name group_id."""
+        group_ids = self.ids
+        self.account_ids = self.env["account.account"]
+        if not group_ids:
+            return
+        group_ids = SQL(",".join(map(str, group_ids)))
+        results = self.env.execute_query(
+            SQL(
+                """
+SELECT
+ agroup.id AS group_id,
+STRING_AGG(DISTINCT account.id::text, ', ') as account_ids
+FROM  account_group agroup
+inner join account_account account
+ON agroup.code_prefix_start <= LEFT(%(code_store)s->>agroup.company_id::text,
+char_length(agroup.code_prefix_start))
+AND agroup.code_prefix_end >= LEFT(%(code_store)s->>agroup.company_id::text,
+char_length(agroup.code_prefix_end))
+AND agroup.id IN (%(group_ids)s)
```

Comment:
```suggestion
WHERE agroup.id IN (%(group_ids)s)
```
I don't know if it changes the result of the query, but it seems reasonable for this to be a WHERE condition, let me know what you think.

Reply:
fixed